src/generators.rs

```diff
@@ -69,12 +71,19 @@ where
 ///   image
 ///
 #[pyfunction]
```

Comment:
I think you should add the kwarg to the definition here so the Python C API FFI function this macro generates sets a default. This will avoid the need to use an `Option<bool>` below. Something like:

```suggestion
#[pyfunction(bidirectional = "false")]
```

This basically makes it so Python callers get an implicit default of `false`.